### ISSUE TITLE
Users receive email when their saved events change

### ISSUE THREAD

**Issue by kps1ny** [skandragon/thing#174]
It would be a great help to receive an email when an event's time, date, or location is changed, or it is canceled.

Given that people can save a set of events, and could opt to provide an email address for their account, it ought to be feasible to arrange for this to happen.

Is the Thing still an active project? If we put the time into making such changes, is there anyone there to pull them?

**Comment by Artimisia** [comment]
> On Aug 14, 2022, at 11:25 PM, Ken Stuart ***@***.***> wrote:
> 
> Is the Thing still an active project? If we put the time into making such changes, is there anyone there to pull them?

Yes. Thing is still an ongoing project.  In fact, we are hoping for a complete, modern rewrite soon.  We wanted it by 50, but life may have happened.

I’m pretty sure what you want is part of the specs on the new version already, along with making it mobile friendly.

If you would like to work on the rewrite or tweak the current one, please let us know.  The current project lead is JJ Ryan - ***@***.***

Gwynnyd

Mistress Gwynnyd, OP, Baroness Roaring Wastes
Registrar, Pennsic University 
Pennsic War 49


***@***.***	
or ***@***.***

----------

Illud Latine dici non possum

**Comment by kps1ny** [comment]
> 

@Artimisia thanks for this reply, it's encouraging. I have been a professional web developer for over 25 years (since NCSA Mosaic) but have not worked with Ruby on Rails. I would love to help in some capacity, but not sure how useful I would be in trying to code in an unfamiliar language. Maybe assist with testing or QA? Or if the new version is in some other language maybe there? -Robert of Ferness, OL

**Comment by jjryan111** [comment]
Wow Mosaic. I’m impressed. We are starting to discuss Thing 2.0. It will not be Ruby on Rails. What is your skillset?

**Comment by jjryan111** [comment]
Is the info on your profile current? Let’s discuss this off github.

**Comment by adigitalnative** [comment]
Hey there, tagging in. I'm a professional RoR dev with some React thrown in. Sounds like you are moving away from Rails but nevertheless might be an extra set of hands.

**Comment by jjryan111** [comment]
It's great to see people who are willing to help. If you all have some time to devote to Thing 2.0 please let me know your level of commitment (in hours/week) for the new project. 

JJ

**Comment by Cordelya** [comment]
Chiming in as well. I can bring Django/Python, php, and CSS. Familiar with git. Willing to learn new frameworks etc. Able to spin up local development environments. I'm unemployed (not looking terribly hard, either) so I could devote quite a few hours per week when needed. I'm ADHD so if it looks like I've disappeared, poke at me. 

Additionally interested in expanding documentation and other tools, not only for Users, but for folks who want to spin up a Thing instance for other events. Have done a bit already with VM-building tools like `cloud-init` for Ubuntu Multipass and Docker.

Known in the Society as Cordeilla Sharpe (Shire of Owlsherst, East)
[personal git repos](https://gitlab.com/Cordelya)
[SCA git repos](https://gitlab.com/223686)

<details>
<summary>Other SCA webministry-related activities - because you won't find all of this on my profile(s)</summary>

- Local Webminister (branch & regional event WordPress sites)
- Dev on EK infrastructure app currently in development
- Maintainer of [Gleann Abhann's OP](https://records.gleannabhann.net) application
- Initiator of _I Cifristi di Ea-Nasir_ ("Ea-Nasir's Encipherers"; a Special Interest Group / Household recently established to grow more Webministers from scratch and promote the ongoing development of technological solutions that support SCA operations. The only requirement for membership is that you participate regularly with a local branch) [mailing list](https://groups.google.com/a/webminister.eastkingdom.org/g/cifristi)

</details>

**Comment by wumpus** [comment]
I'm good for a few hours of week of maintenance, but of course new features take a lot more than that!

**Comment by jjryan111** [comment]
HI,
I'm writing because last year you expressed interest in helping to create a new class scheduling system for Pennsic classes. The project is finally getting off the ground and I was wondering if you were still interested. Let me know.
Pretty please? 
-JJ
 

    On Friday, August 19, 2022 at 09:29:28 PM EDT, Jacqueline Chenault ***@***.***> wrote:  
 
 


Hey there, tagging in. I'm a professional RoR dev with some React thrown in. Sounds like you are moving away from Rails but nevertheless might be an extra set of hands.

**Comment by jjryan111** [comment]
HI,
I'm writing because last year you expressed interest in helping to create a new class scheduling system for Pennsic classes. The project is finally getting off the ground and I was wondering if you were still interested. Let me know.
Pretty please? 
-JJ

    On Friday, August 26, 2022 at 12:44:25 PM EDT, Cordelya Sharpe ***@***.***> wrote:  
 
 


Chiming in as well. I can bring Django/Python, php, and CSS. Familiar with git. Willing to learn new frameworks etc. Able to spin up local development environments. I'm unemployed (not looking terribly hard, either) so I could devote quite a few hours per week when needed. I'm ADHD so if it looks like I've disappeared, poke at me.

Additionally interested in expanding documentation and other tools, not only for Users, but for folks who want to spin up a Thing instance for other events. Have done a bit already with VM-building tools like cloud-init for Ubuntu Multipass and Docker.

Known in the Society as Cordeilla Sharpe (Shire of Owlsherst, East)
personal git repos
SCA git repos
 Other SCA webministry-related activities - because you won't find all of this on my profile(s)    
   - Local Webminister (branch & regional event WordPress sites)
   - Dev on EK infrastructure app currently in development
   - Maintainer of Gleann Abhann's OP application
   - Initiator of I Cifristi di Ea-Nasir ("Ea-Nasir's Encipherers"; a Special Interest Group / Household recently established to grow more Webministers from scratch and promote the ongoing development of technological solutions that support SCA operations. The only requirement for membership is that you participate regularly with a local branch) mailing list

**Comment by adigitalnative** [comment]
Hey there! I should be able to assist but what I can do will depend on what
language/framework you are planning to use.

In my professional life I'm a senior software engineer working full stack
on a Rails/React app (we're currently running Rails 6, I've poked at 7 and
turbo but my company hasn't transitioned over). I'm not really looking to
pick up a new language in my free time right now, so if you are still
planning to use a non-Ruby framework I probably won't be super helpful on
the backend but can still contribute to HTML/CSS, React or JS, etc.

I also have experience with light product work, architecture planning,
database design, roadmaps, and api design so might be able to take some of
the load there.

To summarize:
- If you are working in RoR/Ruby/JS, I'm right there with you and can
probably dive in quickly to start building something that is well tested
and fits the patterns RoR devs know and generally love. I can also tap in
with some project management/feature planning/organizational support.
- If you are working in a different language/framework, I will probably be
most useful working from a more zoomed-out perspective, thinking about
features and organizing work rather than putting hands-on most of the code.
- Either way, I've got enough basic HTML, CSS, and JS to make contributions
on the front-end.

Let me know if any of that sounds helpful and we can go from there!
Jacqueline C./Lady Marguerite du Cheneau, Atlantia

Jacqueline Chenault
***@***.***
202.489.6128

**Comment by kps1ny** [comment]
Pretty much the same for me, but opposite... I have no RoR/Rails experience
and don't have time to pick it up. I could probably help via PHP/Laravel,
as well as Vue, HTML, JavaScript and so on. I'm also a long-time,
full-stack developer, definitely interested in accessibility and usability
aspects of the site.

However, before committing, I'd like to know more details...

Is the plan to have a total or partial overhaul before Pennsic 50? Such a
schedule seems optimistic to me, but if there is a big enough team with
enough time, it might be feasible.

How many people might be involved in the work? Are we all going to put in a
ton of hours trying to get done for Pennsic... and if we don't make it,
have it all be put off for a year? Are we talking about small bits of work
around the edges on the existing system / framework, or starting from
scratch, or what? Also, I assume we are all volunteers, but please correct
me if I'm wrong.

Anyway, an overview of the current situation, the plans, the goals, all of
that stuff would be most helpful in deciding on the level of commitment
needed and where expertise might be provided.

-Ken

**Comment by jjryan111** [comment]
Totally fair.

1) THIS IS NOT FOR PENNSIC 50. Full stop. 51 at the earliest.
2) I'd like to get a team of 4-6 committed devs
3) I'd really like a discussion about frameworks
4) We're starting from scratch so we don't have to worry about carrying legacy stuff into the future. Done enough of that over the last few years. I would like to architect this with planned obsolescence and periodic overhauls in mind.
5) We're all volunteers.

Other questions are welcome. Let me know.

JJ
